### PR TITLE
Adding fix for upper case not supported check for property names

### LINKED ISSUE #294: Using properties with camel case quietly fails
**Version information**
0.1.24

**Describe the bug**
Using camel case for a property quietly doesn't work.
Instead of producing an error when generating python code, it fails when using grpc APIs (e.g. ParseDict) to marshal created objects into a GRPC message.

```
components:
  schemas:
    Thing:
      type: object
      properties:
        thingProperty:
```

Note that it works correctly when using `thing_property`.

**Expected behavior**
If camel case is not supported, a clear error message should be generated when running the openapiart tools in a non-confirming openapi definition.
Or alternatively, camel case should be supported.

### PR DESCRIPTION
**Pull request by dipendughosh**
Fixes #294

### CODE REVIEW

File: openapiart/tests/field_uid/property_name.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      properties:
+        camelCase:
+          type: boolean
+          x-field-uid: 1
+        PascalCase:
+          type: boolean
+          x-field-uid: 2
+        Upper_case:
+          type: boolean
+          x-field-uid: 3
```
Comment (ashutshkumr) [inline openapiart/tests/field_uid/property_name.yaml:64]: Is the test actually exercising all three use cases ?

File: openapiart/tests/field_uid/property_name.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  errors:
+                    type: array
+                    items:
+                      type: string
+                    x-field-uid: 1
```
Comment (ashutshkumr) [inline openapiart/tests/field_uid/property_name.yaml:48]: these are not really needed i guess

Comment (ashutshkumr) [reply openapiart/tests/field_uid/property_name.yaml:48]: just a simple 200 without any payload should be fine

File: openapiart/bundler.py [hunk 3]
```diff
@@ -336,8 +345,15 @@ def _validate_names(self, regex, components_key, components):
                             "%s property name `%s` contains invalid characters"
                             % (key, name)
                         )
+                    upper_case = self._check_upper_case(name)
+                    if upper_case:
+                        raise NameError("*** Property name '{}' is invalid. Only lower case with underscore separated is allowed.***".format(
```
Comment (ashutshkumr) [inline openapiart/bundler.py]: Only lower case with underscore separated is allowed -> Only lower case letters separated with an underscore is allowed

File: openapiart/bundler.py [hunk 4]
```diff
@@ -336,8 +345,15 @@ def _validate_names(self, regex, components_key, components):
                             "%s property name `%s` contains invalid characters"
                             % (key, name)
                         )
+                    upper_case = self._check_upper_case(name)
+                    if upper_case:
```
Comment (ashutshkumr) [inline openapiart/bundler.py:349]: would suggest directly using calling the method here if the value is not needed anywhere else

File: openapiart/bundler.py [hunk 5]
```diff
@@ -323,7 +323,16 @@ def _process_yaml_object(self, base_dir, yobject):
...
-
+    
+    def _check_upper_case(self, value):
+        upper_case = False
+                
+        for c in value:
+            if c.isupper():
+                upper_case = True
+        
+        return upper_case
```
Comment (ashutshkumr) [inline openapiart/bundler.py:334]: would suggest breaking early in the function; probably storing False/True in upper_case var is not required

**Review by ashutshkumr — CHANGES_REQUESTED** [review]
@dipendughosh @Vibaswan kindly check why CI is failing

File: openapiart/tests/field_uid/property_name.yaml (hunk 1 above)
Comment (dipendughosh) [reply openapiart/tests/field_uid/property_name.yaml:64]: Only the 1st one will be tested but the remaining are for future users to understand what we are not supporting.

**Comment by Vibaswan** [comment]
@ashutshkumr 
The CI was failing due a error from flake8
I also faced similar issue and suggested @dipendughosh with the fix I used earlier 
https://github.com/open-traffic-generator/openapiart/commit/f461cf37ac88ebed82d814d091183be030ca4d9f

I guess this fix would also be needed 
https://github.com/open-traffic-generator/openapiart/commit/aef08dfbbed1b9c6d80e83b159cb5878b36cfbe0

File: openapiart/bundler.py (hunk 3 above)
Comment (dipendughosh) [reply openapiart/bundler.py]: done

File: openapiart/bundler.py (hunk 4 above)
Comment (dipendughosh) [reply openapiart/bundler.py]: done

File: openapiart/bundler.py (hunk 5 above)
Comment (ashutshkumr) [reply openapiart/bundler.py]: Can't see the change here

File: openapiart/tests/field_uid/property_name.yaml (hunk 1 above)
Comment (dipendughosh) [reply openapiart/tests/field_uid/property_name.yaml]: done

File: openapiart/bundler.py (hunk 5 above)
Comment (dipendughosh) [reply openapiart/bundler.py]: done

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**